### ISSUE TITLE
JSPM with TextAngular don´t work

### ISSUE THREAD

**Issue by ospar0829** [jspm/jspm-cli#1650]
Hi, i have issues migrating to Jspm with textAngular, I appreciate the help. 

```
TypeError: Unexpected anonymous AMD define.
    at l.reduceRegister_ (http://pruebas.serviradar.com/system-csp-production.js:5:8577)
    at l.pushRegister_ (http://pruebas.serviradar.com/system-csp-production.js:4:29004)
    at a (http://pruebas.serviradar.com/system-csp-production.js:5:8041)
    at http://pruebas.serviradar.com/layer3.profile.js:10443:13
    at Object.invoke (http://pruebas.serviradar.com/app.js:2040:23)
    at http://pruebas.serviradar.com/app.js:11780:30
    at Object.forEach (http://pruebas.serviradar.com/app.js:91:26)
    at _register (http://pruebas.serviradar.com/app.js:11779:19)
    at success (http://pruebas.serviradar.com/app.js:12104:21)
    at processQueue (http://pruebas.serviradar.com/app.js:7169:34)
```

 (http://pruebas.serviradar.com/app.js:2040:23):

```
q.run([function() {
          if ("function" == typeof define && define.amd)
            define(function(a) { **//line 23 **
				window.rangy = a("npm:rangy@1.3.0"), window.rangy.saveSelection = a("npm:rangy@1.3.0/lib/rangy-selectionsaverestore");
            });
          else if ("function" == typeof $__require && "undefined" != typeof module && "object" == typeof a)
            window.rangy = $__require('npm:rangy@1.3.0'), window.rangy.saveSelection = $__require('npm:rangy@1.3.0/lib/rangy-selectionsaverestore');
          else {
            if (!window.rangy)
              throw "rangy-core.js and rangy-selectionsaverestore.js are required for textAngular to work correctly, rangy-core is not yet loaded.";
            if (window.rangy.init(), !window.rangy.saveSelection)
              throw "rangy-selectionsaverestore.js is required for textAngular to work correctly.";
          }
```

apparently the problem is loading rangy, how can I solve it? thanks

**Comment by guybedford** [comment]
Thanks for posting. Is this build file made with the jspm bundle or jspm build command? Can you perhaps share the full file? (the link above doesn't work without a login).

**Comment by guybedford** [comment]
Also the `q.run([function() {` wrapper seems like it might be to blame here. The contents of that are a module, which should be as its own file, not inline along with other module code.

**Comment by ospar0829** [comment]
Thanks @guybedford  .I use a gulp task to make the build process.
```
gulp.task('jspm', function(){
    jspm({
        bundleOptions: {
            minify: false,
            mangle: false
        },
        bundles: [
            { src: 'app/app.module - app/layer2/layer2.module.js - app/core/core.module + angular', dst: 'app.js'},
            { src:  'app/core/core.module - angular', dst: 'lib.js'},
            { src: 'app/layer2/layer2.module.js - angular - angular-ui-router - angular-material - angular-messages - angular-cookies - angular-sanitize' , dst: 'layer2.js' },
            { src: 'app/layer3/layer3.module.js - angular - angular-ui-router - angular-material - angular-messages - angular-cookies - angular-sanitize' , dst: 'layer3.js' },
            { src: 'app/layer3/mario/profile/layer3.profile.module.js - angular - angular-ui-router - angular-material - angular-messages - angular-cookies + angular-sanitize' , dst: 'layer3.profile.js' }
        ],
        configOverride: {
            baseURL: '/'
        }
    })
        .pipe(gulp.dest('./build/'));
}); 
```
I test the others bundles and only 'layer3.profile.js' fail.


```
import 'angular-bootstrap'
import 'angular-slick-carousel'
import 'textangular';
import 'textangular/dist/textAngular-sanitize';
import 'ng-file-upload'
import 'ng-sortable'
import basicInfo from './basic-info/basic-info.controller'
import mediaFiles from './media-files/media-files.controller'
import credentials from './credentials/credentials.directive'
import ProfileMenu from './editprofile.controller'
export default angular.module('app.layer3.profile', [
        'textAngular',
        'ngFileUpload',
        'as.sortable'
    ])
    .directive('basicInfo', basicInfo)
    .directive ('mediaFiles', mediaFiles)
    .directive ('credentials', credentials)
    .controller('ProfileMenu', ProfileMenu)
```

here is the full file. layer3.profile.js:10443:13

[](https://www.dropbox.com/s/9zdbgp0khnvtslc/layer3.profile.js?dl=0 )

**Comment by guybedford** [comment]
@ospar0829 I'm not sure I understand the cause of the issue myself here, but here's what I've got - 

The error message you're getting sounds like it is running the code:

    ```define(function(a) { **//line 23 **
                window.rangy = a("npm:rangy@1.3.0"), window.rangy.saveSelection = a("npm:rangy@1.3.0/lib/rangy-selectionsaverestore");
            });
    ```

You can verify this by adding a console.log just before that define function in the same if statement.

Assuming it is running that path (and please tell me if this is true from testing), then that means that the `define` function is defined. But the top of the wrapper looks like:

```
System.registerDynamic("npm:rangy@1.3.0/lib/rangy-selectionsaverestore", ["npm:rangy@1.3.0/lib/rangy-core"], true, function($__require, exports, module) {
  ;
  var global = this,
      __define = global.define;
  global.define = undefined;
```

Which is exactly designed to remove the global define and ensure define is not defined for the execution duration of the current function. So I'd suggest backtracking from that define statement, to see if and why the `define` is not `undefined` as it should be set by the latter code example.

**Comment by ospar0829** [comment]
Solved in 1.5.1 version of TextAngular
